packages/x-charts/src/internals/components/ChartsAxesGradients/ChartsPiecewiseGradient.tsx

```diff
@@ -13,6 +13,10 @@ type ChartsPiecewiseGradientProps = {
 export default function ChartsPiecewiseGradient(props: ChartsPiecewiseGradientProps) {
   const { isReversed, gradientId, size, direction, scale, colorMap } = props;
 
+  if (size <= 0) {
```

Comment:
Just a not for latter. If we realise of any reason (for example ltr/rtl) that negative size are useful, we could replace that by

```suggestion
  if (size !== 0) {
```
or
```suggestion
  if (size < 0.1 && size > -0.1) {
```